app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreen.kt

```diff
@@ -41,15 +43,26 @@ import com.github.se.signify.ui.common.StatisticsTable
 fun ChallengeHistoryScreen(
     navigationActions: NavigationActions,
     userSession: UserSession,
-    userRepository: UserRepository
+    userRepository: UserRepository,
+    challengeRepository: ChallengeRepository
 ) {
   val userViewModel: UserViewModel =
       viewModel(factory = UserViewModel.factory(userSession, userRepository))
+  val challengeViewModel: ChallengeViewModel =
+      viewModel(factory = ChallengeViewModel.factory(userSession, challengeRepository))
 
   val challengesCompleted = remember { mutableIntStateOf(0) }
   val challengesCreated = remember { mutableIntStateOf(0) }
   val challengesWon = remember { mutableIntStateOf(0) }
-  val pastChallenges by userViewModel.pastChallenges.collectAsState()
+
+  val pastChallengeIds by userViewModel.pastChallenges.collectAsState()
+  LaunchedEffect(pastChallengeIds) { challengeViewModel.getChallenges(pastChallengeIds) }
+  val pastChallenges by challengeViewModel.challenges.collectAsState()
+
+  LaunchedEffect(Unit) {
```

Comment:
I believe you're wrong.
`LaunchedEffect(pastChallengeIds)` should reload challenges every time the challenge list updates, while `LaunchedEffect(Unit)` fetches a user's challenge IDs when the screen is opened. They can't be merged, since the ID fetching operation is asynchronous.

Reply:
You are absolutely right. 
But there is two `LaunchedEffect(Unit)` than can be merged under

  ```
LaunchedEffect(Unit) {
    userViewModel.getFriendsList()
    userViewModel.getPastChallenges()
  }
```
```
LaunchedEffect(Unit) {
    userRepository.getChallengesCompleted(
        userId = userSession.getUserId()!!,
        onSuccess = { completed -> challengesCompleted.intValue = completed },
        onFailure = { exception ->
          Log.e("ChallengeStats", "Failed to fetch challenges completed: ${exception.message}")
......
```

Reply:
Oh, I misinterpreted your comment, my bad.
Try to comment below what you are referencing, it shows up better on Github.

Reply:
Noted !